pkg/cmd/repo/delete/delete.go

```diff
@@ -18,6 +18,7 @@ import (
 
 type DeleteOptions struct {
 	HttpClient func() (*http.Client, error)
+	BaseRepo   func() (ghrepo.Interface, error)
```

Comment:
In the `DeleteOptions` defined on line 28 we will also need to specify 
`BaseRepo: f.BaseRepo` else when we call `opts.BaseRepo()` at line 72 it will be undefined. Unfortunately, with how we do tests, overriding `BaseRepo` this is a hard case to catch since it is defined when running tests but not when running normally.